ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -749,4 +751,23 @@ static NamingStrategy findNamingStrategy(String name, List<NamingStrategy> namin
         }
     }
 
+    @Override
+    public void export(Network network, Properties parameters, DataStore dataStore, String filename) {
+        if (network == null) {
+            throw new IllegalArgumentException("network is null");
+        }
+
+        String namingStrategyName = ConversionParameters.readStringParameter(getFormat(), parameters, NAMING_STRATEGY_PARAMETER, defaultValueConfig);
+        NamingStrategy namingStrategy = findNamingStrategy(namingStrategyName, NAMING_STRATEGY_SUPPLIERS.get());
+
+        UcteNetwork ucteNetwork = createUcteNetwork(network, namingStrategy);
+        UcteDataResolver resolver = new UcteDataResolver();
+        try (OutputStream os = dataStore.newOutputStream(resolver.checkFileExtension(filename) ? filename : Files.getNameWithoutExtension(filename) + ".uct", false);
```

Comment:
Is it really the responsibility of the resolver to check an extension? Should it be a method of the UcteDataFormat as this class knows about the extension?